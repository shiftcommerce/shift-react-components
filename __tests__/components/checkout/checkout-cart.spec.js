// Libraries
import React from 'react'

// Components
import CheckoutCart from '../../../src/components/checkout/checkout-cart'

const updateQuantityfnc = jest.fn()

describe('Checkout Cart', () => {
  test('renders the correct cart summary with single line item', () => {
    // Arrange
    const lineItems = [
      {
        total: 20,
        sub_total: 20,
        total_discount: 0,
        unit_quantity: 2,
        stock_available_level: '1000',
        item: {
          imageUrl: 'https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg',
          sku: '123',
          title: 'Test Variant',
          product: {
            slug: '1',
            canonical_path: '1',
            title: 'Test Product'
          }
        }
      }
    ]

    // Act
    const wrapper = mount(
      <CheckoutCart
        lineItems={lineItems}
        lineItemsCount={1}
        total={20}
        updateQuantity={updateQuantityfnc}
      />
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('a.c-line-items__amount')).toIncludeText('Â£20')
    expect(wrapper.find('h4.c-line-items__details-title')).toIncludeText('Test Product')
  })

  test('renders the correct cart summary with multiple line items', () => {
    // Arrange
    const lineItems = [
      {
        total: 10,
        sub_total: 10,
        total_discount: 0,
        unit_quantity: 2,
        stock_available_level: '1000',
        item: {
          imageUrl: 'https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg',
          sku: '123',
          title: 'Test Variant',
          product: {
            slug: '1',
            canonical_path: '1',
            title: 'Test Product'
          }
        }
      },
      {
        total: 5,
        sub_total: 5,
        total_discount: 0,
        unit_quantity: 1,
        stock_available_level: '1000',
        item: {
          imageUrl: 'https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg',
          sku: '124',
          title: 'Pretend Variant',
          product: {
            slug: '2',
            canonical_path: '1',
            title: 'Pretend Product'
          }
        }
      }
    ]

    // Act
    const wrapper = mount(
      <CheckoutCart
        lineItems={lineItems}
        lineItemsCount={3}
        total={25}
        updateQuantity={updateQuantityfnc}
      />
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('div.c-line-items')).toIncludeText('Test Product')
    expect(wrapper.find('div.c-line-items')).toIncludeText('Pretend Product')
  })
})
