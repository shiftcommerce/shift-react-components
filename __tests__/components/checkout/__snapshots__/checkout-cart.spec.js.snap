// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Cart renders the correct cart summary with multiple line items 1`] = `
<CheckoutCart
  lineItems={
    Array [
      Object {
        "item": Object {
          "imageUrl": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg",
          "product": Object {
            "canonical_path": "1",
            "slug": "1",
            "title": "Test Product",
          },
          "sku": "123",
          "title": "Test Variant",
        },
        "stock_available_level": "1000",
        "sub_total": 10,
        "total": 10,
        "total_discount": 0,
        "unit_quantity": 2,
      },
      Object {
        "item": Object {
          "imageUrl": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg",
          "product": Object {
            "canonical_path": "1",
            "slug": "2",
            "title": "Pretend Product",
          },
          "sku": "124",
          "title": "Pretend Variant",
        },
        "stock_available_level": "1000",
        "sub_total": 5,
        "total": 5,
        "total_discount": 0,
        "unit_quantity": 1,
      },
    ]
  }
  lineItemsCount={3}
  total={25}
>
  <section
    className="c-checkout-cart__header"
  >
    <div
      className="u-float--left c-checkout-cart__table"
    >
      <h1
        className="c-checkout-cart__title"
      >
         Your Shopping Basket 
        <a
          className="c-checkout-cart__amount"
        >
          (
          3
          )
        </a>
      </h1>
      <h4
        className="c-checkout-cart__total"
      >
        £
        25.00
      </h4>
      <div
        className="c-checkout-cart__text"
      >
        <p
          className="c-checkout-cart__description"
        >
          You have 
          <a>
            3
          </a>
           
          <Plural
            className={null}
            count={3}
            showCount={false}
            singular="item"
            style={Object {}}
            zero={null}
          >
            <span
              className={null}
              style={Object {}}
            >
              items
            </span>
          </Plural>
           in your shopping basket
        </p>
        <p
          className="c-checkout-cart__description"
        >
          <Plural
            className={null}
            count={3}
            plural="These"
            showCount={false}
            singular="This"
            style={Object {}}
            zero={null}
          >
            <span
              className={null}
              style={Object {}}
            >
              These
            </span>
          </Plural>
           
          <Plural
            className={null}
            count={3}
            showCount={false}
            singular="item"
            style={Object {}}
            zero={null}
          >
            <span
              className={null}
              style={Object {}}
            >
              items
            </span>
          </Plural>
           will be saved for 48 hours depending on availablility
        </p>
      </div>
    </div>
  </section>
  <LineItems
    lineItems={
      Array [
        Object {
          "item": Object {
            "imageUrl": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg",
            "product": Object {
              "canonical_path": "1",
              "slug": "1",
              "title": "Test Product",
            },
            "sku": "123",
            "title": "Test Variant",
          },
          "stock_available_level": "1000",
          "sub_total": 10,
          "total": 10,
          "total_discount": 0,
          "unit_quantity": 2,
        },
        Object {
          "item": Object {
            "imageUrl": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg",
            "product": Object {
              "canonical_path": "1",
              "slug": "2",
              "title": "Pretend Product",
            },
            "sku": "124",
            "title": "Pretend Variant",
          },
          "stock_available_level": "1000",
          "sub_total": 5,
          "total": 5,
          "total_discount": 0,
          "unit_quantity": 1,
        },
      ]
    }
    lineItemsCount={3}
  >
    <div
      className="c-line-items"
    >
      <div
        className="c-line-items__sections"
        key="123"
      >
        <div
          className="c-line-items__images"
        >
          <Link
            href="/slug?slug=1"
          >
            <a
              href="/slug?slug=1"
            >
              <Image
                alt="Test Variant"
                aria-label="Test Variant"
                className="c-line-items__image"
                key="1"
              />
            </a>
          </Link>
        </div>
        <div
          className="c-line-items__information"
        >
          <div
            className="c-line-items__title"
          >
            <div
              className="c-line-items__details"
            >
              <h4
                className="c-line-items__details-title u-bold"
              >
                Test Product - Test Variant
              </h4>
              <div
                className="c-line-items__details-sku"
              >
                <span />
              </div>
            </div>
            <div
              className="c-line-items__remove"
            >
              <div
                className="c-line-items__delete"
              >
                <a
                  className="c-line-items__delete-button"
                >
                  Delete
                </a>
              </div>
            </div>
          </div>
          <div
            className="c-line-items__quantity"
          >
            <div
              className="c-line-items__quantity-header"
            >
              <span>
                Quantity
              </span>
            </div>
            <span
              className="c-line-items__quantity-amount"
            >
              2
            </span>
          </div>
          <div
            className="c-line-items__amounts"
          >
            <a
              className="c-line-items__amount c-line-items__amount--total"
            >
              £
              10.00
            </a>
          </div>
        </div>
      </div>
      <div
        className="c-line-items__sections"
        key="124"
      >
        <div
          className="c-line-items__images"
        >
          <Link
            href="/slug?slug=1"
          >
            <a
              href="/slug?slug=1"
            >
              <Image
                alt="Pretend Variant"
                aria-label="Pretend Variant"
                className="c-line-items__image"
                key="2"
              />
            </a>
          </Link>
        </div>
        <div
          className="c-line-items__information"
        >
          <div
            className="c-line-items__title"
          >
            <div
              className="c-line-items__details"
            >
              <h4
                className="c-line-items__details-title u-bold"
              >
                Pretend Product - Pretend Variant
              </h4>
              <div
                className="c-line-items__details-sku"
              >
                <span />
              </div>
            </div>
            <div
              className="c-line-items__remove"
            >
              <div
                className="c-line-items__delete"
              >
                <a
                  className="c-line-items__delete-button"
                >
                  Delete
                </a>
              </div>
            </div>
          </div>
          <div
            className="c-line-items__quantity"
          >
            <div
              className="c-line-items__quantity-header"
            >
              <span>
                Quantity
              </span>
            </div>
            <span
              className="c-line-items__quantity-amount"
            >
              1
            </span>
          </div>
          <div
            className="c-line-items__amounts"
          >
            <a
              className="c-line-items__amount c-line-items__amount--total"
            >
              £
              5.00
            </a>
          </div>
        </div>
      </div>
    </div>
  </LineItems>
</CheckoutCart>
`;

exports[`Checkout Cart renders the correct cart summary with single line item 1`] = `
<CheckoutCart
  lineItems={
    Array [
      Object {
        "item": Object {
          "imageUrl": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg",
          "product": Object {
            "canonical_path": "1",
            "slug": "1",
            "title": "Test Product",
          },
          "sku": "123",
          "title": "Test Variant",
        },
        "stock_available_level": "1000",
        "sub_total": 20,
        "total": 20,
        "total_discount": 0,
        "unit_quantity": 2,
      },
    ]
  }
  lineItemsCount={1}
  total={20}
>
  <section
    className="c-checkout-cart__header"
  >
    <div
      className="u-float--left c-checkout-cart__table"
    >
      <h1
        className="c-checkout-cart__title"
      >
         Your Shopping Basket 
        <a
          className="c-checkout-cart__amount"
        >
          (
          1
          )
        </a>
      </h1>
      <h4
        className="c-checkout-cart__total"
      >
        £
        20.00
      </h4>
      <div
        className="c-checkout-cart__text"
      >
        <p
          className="c-checkout-cart__description"
        >
          You have 
          <a>
            1
          </a>
           
          <Plural
            className={null}
            count={1}
            showCount={false}
            singular="item"
            style={Object {}}
            zero={null}
          >
            <span
              className={null}
              style={Object {}}
            >
              item
            </span>
          </Plural>
           in your shopping basket
        </p>
        <p
          className="c-checkout-cart__description"
        >
          <Plural
            className={null}
            count={1}
            plural="These"
            showCount={false}
            singular="This"
            style={Object {}}
            zero={null}
          >
            <span
              className={null}
              style={Object {}}
            >
              This
            </span>
          </Plural>
           
          <Plural
            className={null}
            count={1}
            showCount={false}
            singular="item"
            style={Object {}}
            zero={null}
          >
            <span
              className={null}
              style={Object {}}
            >
              item
            </span>
          </Plural>
           will be saved for 48 hours depending on availablility
        </p>
      </div>
    </div>
  </section>
  <LineItems
    lineItems={
      Array [
        Object {
          "item": Object {
            "imageUrl": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg",
            "product": Object {
              "canonical_path": "1",
              "slug": "1",
              "title": "Test Product",
            },
            "sku": "123",
            "title": "Test Variant",
          },
          "stock_available_level": "1000",
          "sub_total": 20,
          "total": 20,
          "total_discount": 0,
          "unit_quantity": 2,
        },
      ]
    }
    lineItemsCount={1}
  >
    <div
      className="c-line-items"
    >
      <div
        className="c-line-items__sections"
        key="123"
      >
        <div
          className="c-line-items__images"
        >
          <Link
            href="/slug?slug=1"
          >
            <a
              href="/slug?slug=1"
            >
              <Image
                alt="Test Variant"
                aria-label="Test Variant"
                className="c-line-items__image"
                key="1"
              />
            </a>
          </Link>
        </div>
        <div
          className="c-line-items__information"
        >
          <div
            className="c-line-items__title"
          >
            <div
              className="c-line-items__details"
            >
              <h4
                className="c-line-items__details-title u-bold"
              >
                Test Product - Test Variant
              </h4>
              <div
                className="c-line-items__details-sku"
              >
                <span />
              </div>
            </div>
            <div
              className="c-line-items__remove"
            >
              <div
                className="c-line-items__delete"
              >
                <a
                  className="c-line-items__delete-button"
                >
                  Delete
                </a>
              </div>
            </div>
          </div>
          <div
            className="c-line-items__quantity"
          >
            <div
              className="c-line-items__quantity-header"
            >
              <span>
                Quantity
              </span>
            </div>
            <span
              className="c-line-items__quantity-amount"
            >
              2
            </span>
          </div>
          <div
            className="c-line-items__amounts"
          >
            <a
              className="c-line-items__amount c-line-items__amount--total"
            >
              £
              20.00
            </a>
          </div>
        </div>
      </div>
    </div>
  </LineItems>
</CheckoutCart>
`;
